drivers/fuel_gauge/sbs_gauge/sbs_gauge.c

```diff
@@ -307,17 +331,38 @@ static const struct fuel_gauge_driver_api sbs_gauge_driver_api = {
 	.get_property = &sbs_gauge_get_props,
 	.set_property = &sbs_gauge_set_props,
 	.get_buffer_property = &sbs_gauge_get_buffer_prop,
+#ifdef SBS_GAUGE_CUTOFF_SUPPORT
+	.battery_cutoff = &sbs_gauge_do_battery_cutoff,
+#endif
 };
 
-/* FIXME: fix init priority */
+#ifdef SBS_GAUGE_CUTOFF_SUPPORT
+#define SBS_GAUGE_CONFIG_INIT(index)                                                               \
+	.cutoff_support = DT_PROP_OR(DT_DRV_INST(index), battery_cutoff_support, false),           \
+	.cutoff_reg_addr = DT_PROP_OR(DT_DRV_INST(index), battery_cutoff_reg_addr, 0),             \
+	.cutoff_payload = DT_PROP_OR(DT_DRV_INST(index), battery_cutoff_payload, {}),
+#else
+#define SBS_GAUGE_CONFIG_INIT(index)
+#endif
+
 #define SBS_GAUGE_INIT(index)                                                                      \
-                                                                                                   \
 	static const struct sbs_gauge_config sbs_gauge_config_##index = {                          \
-		.i2c = I2C_DT_SPEC_INST_GET(index),                                                \
-	};                                                                                         \
+		.i2c = I2C_DT_SPEC_INST_GET(index), SBS_GAUGE_CONFIG_INIT(index)};                 \
                                                                                                    \
 	DEVICE_DT_INST_DEFINE(index, &sbs_gauge_init, NULL, NULL, &sbs_gauge_config_##index,       \
 			      POST_KERNEL, CONFIG_FUEL_GAUGE_INIT_PRIORITY,                        \
 			      &sbs_gauge_driver_api);
 
 DT_INST_FOREACH_STATUS_OKAY(SBS_GAUGE_INIT)
+
+
+/* Validate that all cutoff payloads no larger than size 2 */
+
+#define _SBS_GAUGE_VALIDATE_CUTOFF_PAYLOAD_SIZE(index)                                             \
+	&&(DT_PROP_LEN_OR(DT_DRV_INST(n), battery_cutoff_payload, 0) <=                            \
+	   SBS_GAUGE_CUTOFF_PAYLOAD_MAX_SIZE)
+
+#define SBS_GAUGE_VALIDATE_CUTOFF_PAYLOAD_SIZE(index)                                              \
+	(true _SBS_GAUGE_VALIDATE_CUTOFF_PAYLOAD_SIZE(index))
+
+BUILD_ASSERT(DT_INST_FOREACH_STATUS_OKAY(SBS_GAUGE_VALIDATE_CUTOFF_PAYLOAD_SIZE));
```

Comment:
This can be simplified as:
```
#define CUTOFF_PAYLOAD_SIZE_ASSERT(inst) \
    BUILD_ASSERT(DT_INST_PROP_OR(inst, battery_cutoff_payload, 0) <=   \
        SBS_GAUGE_CUTOFF_PAYLOAD_MAX_SIZE)
DT_INST_FOREACH_STATUS_OKAY(CUTOFF_PAYLOAD_SIZE_ASSERT)
```